src/Discovery/DiscoveryActionBar/DiscoveryDataLibrary.tsx

```diff
@@ -0,0 +1,160 @@
+    }
+    }
+  } catch (error) {
+    postNotification('Unknown error. Please try again later.', 'error');
+  }
+};
+
+/* eslint react/prop-types: 0 */
+interface Props {
+    config: DiscoveryConfig;
+    // eslint-disable-next-line react/no-unused-prop-types
+    user: User;
+    discovery: {
+        selectedResources: any[];
+    };
+  healIDPLoginNeeded: string[];
+}
+
+const DiscoveryDataLibrary = (props: Props) => {
+  const [data, setData] = useState<ComboboxItem[]>([]);
+  const [currentList, setCurrentList] = useState<ComboboxItem>(undefined);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+  const { config: { features: { exportToDataLibrary } }, discovery: { selectedResources }, healIDPLoginNeeded } = props;
+
+  const onChangeListSelection = (listname: string, listId: string) => {
+    setCurrentList({ label: listname, value: listId });
+  };
+
+  const saveToList = (listname: string, listId: string = undefined) => {
+    if (selectedResources.length === 0) return;
+    const items = selectedResources.reduce((acc, resource) => {
+      const dataObjects = resource[exportToDataLibrary.dataObjectFieldName];
+      const datasetId = resource[exportToDataLibrary.datesetIdFieldName];
+
+      const datafiles = dataObjects.reduce((dataAcc, dataObject: any) => {
+        const guid = dataObject[exportToDataLibrary.dataObjectIdField];
+        return {
+          ...dataAcc,
+          [guid]: {
+            dataset_guid: datasetId,
+            ...dataObject,
+          },
+        };
+      }, {});
+      return {
+        ...acc,
+        ...datafiles,
+      };
+    }, {});
+
+    try {
+      updateList({
+        name: listname,
+        items,
+      }, listId);
+    } catch (err) {
+      postNotification('Data Library update failed. Please try again later.', 'error');
+    }
+  };
+
+  // fetch the list
+  const fetchLists = async () => {
+    setLoading(true);
+    setError(null);
+
+    try {
+      const response = await fetchWithCreds({ path: `${userDataLibraryUrl}/lists` });
+      const jsonData = response.data;
+      const listItems = extractListNameAndId(jsonData.lists);
+      setData(listItems);
+    } catch (err) {
+      setError(err.message);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchLists();
+  }, []);
+
+  const notLoggedIn = !(props.user.username && !(healIDPLoginNeeded.length > 0));
+
+  return (
+    <React.Fragment>
+      <Space size='small'>
+        <ComboboxWithInput
+          items={data}
+          onChange={onChangeListSelection}
+          placeholder={notLoggedIn ? 'Login to save to a list' : 'Select/create a list to save to'}
+          disabled={notLoggedIn}
+        />
+        <Button
+          loading={loading}
+          type='primary'
+          disabled={error !== null || loading || data?.length === 0 || currentList === undefined || selectedResources.length === 0 || notLoggedIn}
```

Comment:
`data?.length === 0` seems to prevent any initial list creation. It does not seem to increase when a user's first list is initially entered (after creation, but before a study is Saved to list), and subsequently can't be Saved to List as the button remains disabled. I have tested removing this and it resolves [this issue](https://ctds-planx.atlassian.net/browse/PXP-11460)